In [3]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker im
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

ModuleNotFoundError: No module named 'numpy'

In [5]:
import numpy as np
import pandas as pd

ModuleNotFoundError: No module named 'numpy'

In [4]:
df = pd.read_csv("/kaggle/input/european-restaurant-reviews/European Restaurant Reviews.csv")

NameError: name 'pd' is not defined

In [ ]:
df.head()

Country            Restaurant Name Sentiment  \
0  France  The Frog at Bercy Village  Negative   
1  France  The Frog at Bercy Village  Negative   
2  France  The Frog at Bercy Village  Negative   
3  France  The Frog at Bercy Village  Negative   
4  France  The Frog at Bercy Village  Negative   

                                Review Title Review Date  \
0                               Rude manager  May 2024 •   
1                       A big disappointment  Feb 2024 •   
2               Pretty Place with Bland Food  Nov 2023 •   
3   Great service and wine but inedible food  Mar 2023 •   
4  Avoid- Worst meal in Rome - possibly ever  Nov 2022 •   

                                              Review  
0  The manager became agressive when I said the c...  
1  I ordered a beef fillet ask to be done medium,...  
2  This is an attractive venue with welcoming, al...  
3  Sadly I  used the high TripAdvisor rating too ...  
4  From the start this meal was bad- especially g...

In [ ]:
df.tail()

Country           Restaurant Name Sentiment  \
1497    Cuba  Old Square (Plaza Vieja)  Negative   
1498    Cuba  Old Square (Plaza Vieja)  Negative   
1499    Cuba  Old Square (Plaza Vieja)  Negative   
1500    Cuba  Old Square (Plaza Vieja)  Negative   
1501    Cuba  Old Square (Plaza Vieja)  Negative   

                          Review Title Review Date  \
1497                  The Tourism Trap  Oct 2016 •   
1498                  the beer factory  Oct 2016 •   
1499                           brewery  Oct 2016 •   
1500  It's nothing exciting over there  Oct 2016 •   
1501                      Tourist Trap  Oct 2016 •   

                                                 Review  
1497  Despite the other reviews saying that this is ...  
1498  beer is good.  food is awfull  The only decent...  
1499  for terrible service of a truly comedic level,...  
1500  We visited the Havana's Club Museum which is l...  
1501  Food and service was awful. Very pretty stop. ...

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1502 entries, 0 to 1501
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Country          1502 non-null   object
 1   Restaurant Name  1502 non-null   object
 2   Sentiment        1502 non-null   object
 3   Review Title     1502 non-null   object
 4   Review Date      1502 non-null   object
 5   Review           1502 non-null   object
dtypes: object(6)
memory usage: 70.5+ KB


In [ ]:
df.describe()

Country            Restaurant Name Sentiment Review Title Review Date  \
count     1502                       1502      1502         1502        1502   
unique       7                          7         2         1343         143   
top     France  The Frog at Bercy Village  Positive    Excellent  May 2014 •   
freq       512                        512      1237            9         108   

                                                   Review  
count                                                1502  
unique                                               1426  
top     I'm afraid we were very let down by this resta...  
freq                                                    4

In [ ]:
df.isnull().sum()

Country            0
Restaurant Name    0
Sentiment          0
Review Title       0
Review Date        0
Review             0
dtype: int64

In [ ]:
df.sample(5)

Country                        Restaurant Name Sentiment  \
992    India                                 Mosaic  Positive   
725    Italy  Ad Hoc Ristorante (Piazza del Popolo)  Positive   
116   France              The Frog at Bercy Village  Negative   
792    Italy  Ad Hoc Ristorante (Piazza del Popolo)  Positive   
1362    Cuba               Old Square (Plaza Vieja)  Positive   

                                           Review Title Review Date  \
992              Diner and breakfast with my colleagues  Jun 2024 •   
725                                             Lovely!  Apr 2019 •   
116   A boring four and a half hours that could have...  Oct 2011 •   
792                                      Amazing dinner  Nov 2018 •   
1362                                        Nice Square  May 2014 •   

                                                 Review  
992   I liked the diner it was very delicious and al...  
725   I don't know how we managed to miss this on pr...  
116   We arrived at 08.15 with table booked for 08.3...  
792   My wife and I visited "Ad hoc" in November 201...  
1362  Nice place, many choices for lunch or dinner. ...

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
sns.set_theme(style="darkgrid")
import warnings
warnings.filterwarnings('ignore')

In [ ]:
country_countwise = df[['Country']].value_counts().sort_values(ascending=False)

In [ ]:
df_country_countwise= country_countwise.reset_index()
df_country_countwise

Country  count
0   France    512
1    Italy    318
2  Morroco    210
3     Cuba    146
4   Poland    135
5   Russia    100
6    India     81

In [ ]:
# Count of Reviews given by countries

fig = px.bar(df_country_countwise, x='Country', y='count', color = 'Country')
fig.show()


Reviews given by France is maximum and reviews given by India is least.

In [ ]:
sentiment_count = df['Sentiment'].value_counts()
df_sentiment_count = sentiment_count.reset_index()
df_sentiment_count

Sentiment  count
0  Positive   1237
1  Negative    265

In [ ]:
# Analysis of sentiments

fig = px.pie(df_sentiment_count, values='count', names='Sentiment',color = 'Sentiment',color_discrete_map = {'Positive':'lightcyan','Negative':'cyan'})
fig.show()

As compared to Negative sentiments, Positive sentiments are in majority among reviews.

In [ ]:
df['Restaurant Name'].unique()

array(['The Frog at Bercy Village',
       'Ad Hoc Ristorante (Piazza del Popolo)', 'Stara Kamienica',
       'Mosaic', 'Pelmenya', 'The LOFT', 'Old Square (Plaza Vieja)'],
      dtype=object)

In [ ]:
restaurant_count_of_country = df[['Restaurant Name']].value_counts()
restaurant_count_of_country

Restaurant Name                      
The Frog at Bercy Village                512
Ad Hoc Ristorante (Piazza del Popolo)    318
The LOFT                                 210
Old Square (Plaza Vieja)                 146
Stara Kamienica                          135
Pelmenya                                 100
Mosaic                                    81
Name: count, dtype: int64

In [ ]:
# Restaurant Names in different countries

Restaurants_in_countries =  df.groupby('Country')[['Restaurant Name']].value_counts().reset_index()
Restaurants_in_countries

Country                        Restaurant Name  count
0     Cuba               Old Square (Plaza Vieja)    146
1   France              The Frog at Bercy Village    512
2    India                                 Mosaic     81
3    Italy  Ad Hoc Ristorante (Piazza del Popolo)    318
4  Morroco                               The LOFT    210
5   Poland                        Stara Kamienica    135
6   Russia                               Pelmenya    100

In [ ]:
# Restaurant vs Review Sentiments

#1.Restaurant = 'Old Square (Plaza Vieja)'

Oldsquare = df[(df['Restaurant Name']== 'Old Square (Plaza Vieja)')]
Oldsquare['Sentiment'].value_counts()

Sentiment
Positive    109
Negative     37
Name: count, dtype: int64

Restaurant "Old Square (Plaza Vieja)" got almost 75% positive sentiment reviews and 25% negative sentiment reviews.

In [ ]:
#2.Restaurant = 'The Frog at Bercy Village'

TheFrogatBercyVillage = df[(df['Restaurant Name']== 'The Frog at Bercy Village')]
TheFrogatBercyVillage['Sentiment'].value_counts()

Sentiment
Positive    360
Negative    152
Name: count, dtype: int64

Restaurant got 30% negative sentiment reviews and 70% positive sentiment reviews.

In [ ]:
#3.Restaurant = 'Mosaic'

Mosaic = df[(df['Restaurant Name']== 'Mosaic')]
Mosaic['Sentiment'].value_counts()

Sentiment
Positive    81
Name: count, dtype: int64

Restaurant 'Mosaic' got only positive sentiment review

In [ ]:
#4.Restaurant = 'Ad Hoc Ristorante (Piazza del Popolo)'

AdHocRistorante = df[(df['Restaurant Name']== 'AdHocRistorante')]
AdHocRistorante['Sentiment'].value_counts()

Series([], Name: count, dtype: int64)

This restaurant got neither positive nor negative sentiment review.

In [ ]:
#5.Restaurant = 'The LOFT'

TheLOFT = df[(df['Restaurant Name']== 'The LOFT')]
TheLOFT['Sentiment'].value_counts()

Sentiment
Positive    207
Negative      3
Name: count, dtype: int64

Restaurant 'The LOFT' got 207 positive sentiment review and only 3 negative sentiment review.

In [ ]:
#6.Restaurant = 'Stara Kamienica'

StaraKamienica = df[(df['Restaurant Name']== 'Stara Kamienica')]
StaraKamienica['Sentiment'].value_counts()

Sentiment
Positive    120
Negative     15
Name: count, dtype: int64

Restaurant 'Stara Kamienica' got 89% positive sentiment review and 11% negtive sentiment review.

In [ ]:
#7.Restaurant = 'Pelmenya'

Pelmenya = df[(df['Restaurant Name']== 'Pelmenya')]
Pelmenya['Sentiment'].value_counts()

Sentiment
Positive    90
Negative    10
Name: count, dtype: int64

Restaurant 'Pelmenya' got 90% positive sentiment review and 10% negative sentiment review.

Analysis : As per the above anlaysis of restaurants based on sentiments, we can say that 'Mosaic' is the best restaurant as it has inky positive sentiment reviews.